### PR TITLE
NAS-100846: rewrite replication documentation

### PR DESCRIPTION
**Pull request by Mrt134**
Large changes to document the new replication system behaviors and workflows, changes to periodic snapshot tasks, and adding new SSH Connections and SSH Keypairs screens to the System menu.

This is a DRAFT, extensive testing and refinement is required. The procedures for setting up a replication with a Manually configured SSH Connection are tested and working, but everything related to Semi-automatic SSH connections appear to be broken. Nothing was touched for the TrueNAS specific sections; that text will need to be closely examined later.

This PR addresses docs needs in these Jira tickets:
[NAS-100846](https://jira.ixsystems.com/browse/NAS-100846) (Epic)
  [NAS-101285](https://jira.ixsystems.com/browse/NAS-101285)
  [NAS-101276](https://jira.ixsystems.com/browse/NAS-101276)
  [NAS-101272](https://jira.ixsystems.com/browse/NAS-101272)
  [NAS-101278](https://jira.ixsystems.com/browse/NAS-101278)
  [NAS-101274](https://jira.ixsystems.com/browse/NAS-101274)
  [NAS-101273](https://jira.ixsystems.com/browse/NAS-101273)
  [NAS-101277](https://jira.ixsystems.com/browse/NAS-101277)
  [NAS-101284](https://jira.ixsystems.com/browse/NAS-101284)
  [NAS-100985](https://jira.ixsystems.com/browse/NAS-100985)
  [NAS-101108](https://jira.ixsystems.com/browse/NAS-101108)
  [NAS-101109](https://jira.ixsystems.com/browse/NAS-101109)
  [NAS-101110](https://jira.ixsystems.com/browse/NAS-101110)
    [NAS-101177](https://jira.ixsystems.com/browse/NAS-101177)
  [NAS-101111](https://jira.ixsystems.com/browse/NAS-101111)
  [NAS-101265](https://jira.ixsystems.com/browse/NAS-101265)

These commits will not be cherry-picked to master branch. Legacy UI changes are being tracked in separate tickets and must be addressed separately.

### CODE REVIEW

File: userguide/tasks.rst [hunk 1]
```diff
@@ -757,51 +757,53 @@ describes the fields in this screen.
...
-to exclude particular child  datasets from a recursive snapshot.
-
 Click :guilabel:`SAVE` when finished customizing the task. Defined tasks
-are listed alphabetically in :guilabel:`Periodic Snapshot Tasks`. Click
-|ui-options| for an entry to display the :guilabel:`Edit` and
-:guilabel:`Delete` buttons.
+are listed alphabetically in
+:menuselection:`Tasks --> Periodic Snapshot Tasks`.
+
+Click |ui-options| for an periodic snapshot task to see options to
```
Comment (a-stjohn) [inline userguide/tasks.rst]:
```suggestion
Click |ui-options| for a periodic snapshot task to see options to
```

File: userguide/tasks.rst [hunk 2]
```diff
@@ -811,453 +813,340 @@ Replication Tasks
...
-replications only send the changes to the destination system.
-Interrupting a running replication requires the replication task to
-restart from the beginning.
-
-The target dataset on the receiving system is automatically created in
+The basic configuration requires a source system with the original
+data, a destination system that will store copied data, and an
+:ref:`SSH Connection <SSH Connections>` to establish the link between
+both systems. When an active
+:ref:`periodic snapshot <Periodic Snapshot Tasks>` schedule takes
```
Comment (a-stjohn) [inline userguide/tasks.rst]:
```suggestion
:ref:`periodic snapshot task <Periodic Snapshot Tasks>` takes
```

File: userguide/tasks.rst [hunk 3]
```diff
@@ -811,453 +813,340 @@ Replication Tasks
...
-   zfs allow -ldu repluser create,destroy,diff,mount,readonly,receive,release,send,userprop betapool/alphadata
-
-
-The destination dataset must also be set to read-only. Enter this
-command in the :ref:`Shell`:
-
-.. code-block:: none
-
-   zfs set readonly=on betapool/alphadata
+To begin creating a new replication, go to
```
Comment (a-stjohn) [inline userguide/tasks.rst]:
```suggestion
To create a new replication, go to
```

File: userguide/tasks.rst [hunk 4]
```diff
@@ -1277,54 +1165,17 @@ in data.
...
 Limiting Replication Times
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The :guilabel:`Begin` and :guilabel:`End` times in a replication task
-make it possible to restrict when replication is allowed. These times
-can be set to only allow replication after business hours, or at other
-times when disk or network activity will not slow down other
-operations like snapshots or :ref:`Scrub Tasks`. The default settings
-allow replication to occur at any time.
+The :guilabel:`Schedule` :guilabel:`Begin` and :guilabel:`End` times in
```
Comment (a-stjohn) [inline userguide/tasks.rst]:
```suggestion
The :guilabel:`Schedule`, :guilabel:`Begin`, and :guilabel:`End` times in
```

File: userguide/tasks.rst [hunk 5]
```diff
@@ -811,453 +813,340 @@ Replication Tasks
...
+   |                           |           |                | destination system. Choosing *PULL* hides the :guilabel:`Periodic Snapshot Tasks` field and renames             |
+   |                           |           |                | :guilabel:`Also Include Naming Schema` to :guilabel:`Naming Schema`.                                            |
+   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+
+   | Transport                 | ALL       | drop-down menu | Method of snapshot transfer:                                                                                    |
+   |                           |           |                |                                                                                                                 |
+   |                           |           |                | * *SSH* is supported by most systems. It requires previously created :ref:`SSH Connections`.                    |
```
Comment (wblock) [inline userguide/tasks.rst]: Change to singular "a previously-created SSH connection"

File: userguide/tasks.rst [hunk 6]
```diff
@@ -811,453 +813,340 @@ Replication Tasks
...
+   |                           |           |                | * *SSH+NETCAT* uses SSH to establish a connection to the destination system, then uses                          |
+   |                           |           |                |   `nc(1) <https://www.freebsd.org/cgi/man.cgi?query=nc>`__ to send an unencrypted data stream for higher        |
+   |                           |           |                |   transfer speeds. This is only an option when replicating to a FreeBSD system that has                         |
```
Comment (wblock) [inline userguide/tasks.rst]: As above: define "only an option" to be more clear.  I suspect it means that the user can choose this, but it only works when there is py-libzfs on the remote end.  Not FreeBSD specific, either, could be FreeNAS or TrueNAS or maybe anything with py-libzfs.

File: userguide/tasks.rst [hunk 7]
```diff
@@ -811,453 +813,340 @@ Replication Tasks
...
+   |                           |           |                |   `py-libzfs <https://github.com/freenas/py-libzfs>`__ installed.                                               |
+   |                           |           |                | * *LOCAL* replicates snapshots to another dataset on the same system.                                           |
+   |                           |           |                | * *LEGACY* uses the legacy replication engine from FreeNAS 11.2 and earlier.                                    |
```
Comment (wblock) [inline userguide/tasks.rst]:
```suggestion
   |                           |           |                | * *LEGACY* uses the legacy replication engine from %brand% 11.2 and earlier.                                    |
```

File: userguide/tasks.rst [hunk 8]
```diff
@@ -811,453 +813,340 @@ Replication Tasks
...
+   |                           |           |                |                                                                                                                 |
+   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+
+   | SSH Connection            | SSH, NCT, | drop-down menu | Choose the SSH connection to use for the replication. Choose from a list of connections configured in           |
```
Comment (wblock) [inline userguide/tasks.rst]:
```suggestion
   | SSH Connection            | SSH, NCT, | drop-down menu | Choose the :ref:`SSH connection <SSH Connections>`.          |
```

File: userguide/tasks.rst [hunk 9]
```diff
@@ -811,453 +813,340 @@ Replication Tasks
...
+   |                           | LEG       |                | :ref:`SSH Connections`.                                                                                         |
```
Comment (wblock) [inline userguide/tasks.rst]:
```suggestion
   |                           | LEG       |                |                                                                                          |
```

File: userguide/tasks.rst [hunk 10]
```diff
@@ -811,453 +813,340 @@ Replication Tasks
...
+   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+
+   | Netcat Active Side        | NCT       | drop-down menu | System that will open TCP ports for the direct connection.                                                      |
+   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+
+   | Netcat Active Side Listen | NCT       | string         | Network IP address that the :guilabel:`Active Side` binds to. Defaults to :literal:`0.0.0.0`.                   |
```
Comment (wblock) [inline userguide/tasks.rst]: This whole "active side" part is not clearly defined.

File: userguide/tasks.rst [hunk 11]
```diff
@@ -811,453 +813,340 @@ Replication Tasks
...
+   | Snapshot Replication      | SSH, NCT, | checkbox and   | Schedule which periodic snapshots are replicated. All snapshots are replicated by default. To choose which      |
+   | Schedule                  | LOC       | drop-down menu | snapshots are replicated, set the checkbox and select a schedule from the drop-down menu. For example, the      |
+   |                           |           |                | periodic snapshot task takes a snapshot every hour, but only every other snapshot is needed for replication.    |
+   |                           |           |                | The scheduler is set to even hours and only snapshots taken at those times are replicated.                      |
+   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+
+   | Begin                     | SSH, NCT, | drop-down menu | Define a starting time when the replication cannot run. A replication that is in progress will continue to run  |
```
Comment (wblock) [inline userguide/tasks.rst]: s/will/can/

File: userguide/tasks.rst [hunk 12]
```diff
@@ -811,453 +813,340 @@ Replication Tasks
...
+   |                           | LOC       |                | past this time.                                                                                                 |
+   +---------------------------+-----------+----------------+-----------------------------------------------------------------------------------------------------------------+
+   | End                       | SSH, NCT, | drop-down menu | Define an ending time for the replication restriction.                                                          |
```
Comment (wblock) [inline userguide/tasks.rst]: What restriction?

File: userguide/tasks.rst (hunk 4 above)
Comment (wblock) [reply userguide/tasks.rst]: I think that Begin and End are part of the Schedule, not three different items.

File: userguide/system.rst [hunk 13]
```diff
@@ -1399,6 +1403,184 @@ More details about individual :guilabel:`Provider` settings are
...
+%brand% offers a special semi-automatic setup mode that simplifies
+setting up the SSH connection. On the local system, go to
+:menuselection:`System --> SSH Keypairs`
+and create a new :ref:`SSH Keypair <SSH Keypairs>`.
+Now go to
+:menuselection:`System --> SSH Connections`
+and click |ui-add|.
+
+Choose *Semi-automatic* for the :guilabel:`Setup Method`. Enter the
+hostname or IP address of the remote system in :guilabel:`FreeNAS URL`
```
[the remark this answers is not in the document]
Comment (Mrt134) [reply userguide/system.rst]: Don't have a choice until the UI is updated.

File: userguide/tasks.rst [hunk 14]
```diff
@@ -811,453 +813,340 @@ Replication Tasks
...
-datasets, enabling SSH, and copying encryption keys will vary when the
-destination computer is not a %brand% system.
+To have this replication run immediately after the chosen periodic
+snapshot schedule is complete, set :guilabel:`Run Automatically`.
 
+The :guilabel:`Snapshot Retention Policy` is used to define when
+snapshots are deleted from the destination system. *Same as Source*
+duplicates the snapshot lifetime setting from the source system.
+*Custom* allows defining a snapshot lifetime for the destination system.
+*None* never deletes snapshots from the destination system.
```
[the remark this answers is not in the document]
Comment (Mrt134) [reply userguide/tasks.rst]: UI team needs tickets to address this.

File: userguide/tasks.rst (hunk 6 above)
Comment (Mrt134) [reply userguide/tasks.rst]: The middleware notes about SSH+NETCAT included this caution about FreeBSD systems with py-libzfs. I'll unify this text with the changes I made to the similar text above.

File: userguide/tasks.rst (hunk 10 above)
Comment (Mrt134) [reply userguide/tasks.rst]: yep

File: userguide/tasks.rst (hunk 4 above)
Comment (Mrt134) [reply userguide/tasks.rst]: They are related items, but appear as individual fields. Could go either way on this one.

**Pull request merged**